.gitignore

```diff
@@ -132,3 +132,6 @@ dmypy.json
 
 # Pyre type checker
 .pyre/
+
+# VS Code settings
+.vscode/
```

Comment:
@lowlydba because it's already in the git index, it will also need to specifically removed from git before it will be ignored again.

@Andersson007 Is it necessary to remove the `.vscode/` folder? Checking the settings included in there, it's an extension recommendation for the Ansible vscode extension (folks who open the project will get a message saying that the project recommends installing it, but it won't be forced), and some basic settings enforcing a few whitespace rules.

If they definitely should be removed that's ok just confirming because these are not "temporary files" created by an editor, more akin to an `.editorconfig` type file.

Reply:
@briantist @lowlydba I'm not sure. I think many people don't use vscode and even if they use they can use different settings. I would remove it as even if it's not temporary, it feels like something individual.
Let's ask someone else, @mariolenz @felixfontein what don you think?

Reply:
I found [Should I commit the .vscode folder to source control?](https://stackoverflow.com/q/32964920) with some arguments. Spoiler: There are different opinions.

I don't know VS Code but I remember I ran into some issues with IDE specific configuration in git repositories in the past. Like I ran an older version of the IDE that didn't understand all the settings, or ran a newer version that updated some settings automatically to a new format (breaking the project for people using an older version) or some settings needed an additional extension which I didn't have installed.

If a team of developers all use VS Code (same version with the same extensions installed), it might make sense to share settings. But you don't know what IDE or editor future contributers will use. So I tend to removing it.

Reply:
I agree with @mariolenz: there are a lot of different opinions on this matter. The most important part IMO is that the settings do not unnecessarily restrict contributors, or even make it very hard to contribute since they contain something that is invalid in many cases (a classic case would be paths that only work on some systems, or even just one person's computer).

Regarding having some files in `.vscode/`, and at the same time having `.vscode` in `.gitignore`: that's a common pattern, saying basically "the files we include should be as we include them, and everything else should never be added to this repo".

(I'm also not using vscode, so no idea how useful / annoying the included files are :) )

Reply:
I am using vscode for everything Ansible (and most other things).

I think @felixfontein 's description about including some files (the existing ones), while keeping it .gitignored so that other files are ignored, is the best option actually.

As I mentioned, [`extensions.json`](https://github.com/lowlydba/lowlydba.sqlserver/blob/main/.vscode/extensions.json) is purely a recommendation for the user who opens this repository, suggesting extension(s) that might be helpful. In my opinion, there's no harm at all to keep this one.

The settings in [`settings.json`](https://github.com/lowlydba/lowlydba.sqlserver/blob/main/.vscode/settings.json) are all rather basic (and are all settings I personally use and would encourage):

```json
{
    "files.eol": "\n",
    "files.insertFinalNewline": true,
    "files.trimFinalNewlines": true,
    "files.trimTrailingWhitespace": true
}
```

While users can set additional settings in their user config, workspace settings take precedence so if there's worry about it, then I guess I'd say to remove this one and let users configure these on their own, but these will prevent issues that will cause sanity failures for contributors, and are easy to miss (trailing whitespace for example).

Thoughts based on the above? @Andersson007 @mariolenz @felixfontein @lowlydba

Reply:
how about moving the file to gist or a separate repo and putting a note `if you use vscode, please consider ...` and a link in README?

Reply:
I like keeping as-is (ignored, but present) since they're primarily there to provide a faster feedback cycle on some sanity formatting checks. I think this lowers the bar to contributing by taking out some guesswork for style choices.

The gist feels like a more obtuse in-between state, so if the former isn't workable I'll probably remove the dir entirely.

Reply:
Following the discussion i can see that there are 3 opinions for removing the directory vs 2 for keeping it.
We understand that it can be useful for someone but there's the requirement about unnecessary files + majority of opinions in this discussion.
We shouldn't ignore the majority.
If you think it needs a broader discussion and a Steering Committee vote, feel free to create a [community-topics](https://github.com/ansible-community/community-topics/issues) (heads up: it'll delay the inclusion for up to several weeks but if it's not an issue, we can go through it, no problem).

Reply:
I'd like to keep the `.vscode/` files in *some* capacity - will wait on the discussion to mature before merging in the hopes that this can stay as-is in the current PR state.

Reply:
i won't probably be around next week but will be happy with what the majority thinks:)

Reply:
I have some time to work on this now, so based on the discussion thus far am going to merge this change and will do another PR if needed to modify further. Thanks for the input everyone.